homeassistant/components/sms/__init__.py

```diff
@@ -93,6 +94,17 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
+    # set up notify platform, no entry support for notify component yet,
+    # have to use discovery to load platform.
+    hass.async_create_task(
+        discovery.async_load_platform(
+            hass,
+            Platform.NOTIFY,
+            DOMAIN,
+            {CONF_NAME: DOMAIN},
+            hass.data[DOMAIN],
```

Comment:
We need to pass the original config from `async_setup` as last parameter.

https://github.com/home-assistant/core/blob/0d45cef6f6adce870d0967232fde5a244af958be/homeassistant/components/webostv/__init__.py#L74-L78
https://github.com/home-assistant/core/blob/0d45cef6f6adce870d0967232fde5a244af958be/homeassistant/components/webostv/__init__.py#L108-L119

Reply:
This is code from webostv, I still dont understand what am I missing.

Reply:
Store the original config in hass.data and then pass the original config to `discovery.async_load_platform` as seen here in the webostv integration.

Reply:
done